python/src/spark_rapids_ml/regression.py

```diff
@@ -89,10 +89,21 @@ def _param_mapping(cls) -> Dict[str, Optional[str]]:
         }
 
     @classmethod
-    def _param_value_mapping(cls) -> Dict[str, Dict[str, Union[str, None]]]:
+    def _param_value_mapping(
+        cls,
+    ) -> Dict[str, Callable[[str], Union[None, str, float, int]]]:
         return {
-            "loss": {"squaredError": "squared_loss", "huber": None},
-            "solver": {"auto": "eig", "normal": "eig", "l-bfgs": None},
+            "loss": lambda x: {
+                "squaredError": "squared_loss",
+                "huber": None,
+                "squared_loss": "squared_loss",
```

Comment:
Do we need to map "squared_loss" to "squared_loss"

Reply:
Think this is now required since the original check against `supported_values` has been replaced with the function, which needs to map every possible input to an output now.

Reply:
That's correct.  The previous logic with dicts allowed specifying the mapped output values as input in which case they would be passed through, but with the more general function based logic, it is no longer possible to determine all possible outputs of the function.

Reply:
We should probably update the pydocs for `_param_value_mapping` in core.py to clarify this (requiring all valid inputs to be handled/mapped).

Reply:
Check just updated pydocs.